### PR TITLE
Change From<Handle<A>> for AssetId<A> implementations to implement Borrow<Handle<A>> instead

### PR DESCRIPTION
**Pull request by doonv**
# Objective

In my project I used to do `images.get_mut(mutable_reference_to_handle)`, but that no longer works in 0.12.0 because `From<&mut Handle<A>>` isn't implemented for `AssetId<A>`. This PR fixes that.

## Solution

Replace the implementations:
-  `From<Handle<A>>` for `AssetId<A>`
-  `From<&Handle<A>>` for `AssetId<A>`

With: `impl<T: Borrow<Handle<A>>, A: Asset> From<T> for AssetId<A> `

Result: You can now pass mutable references to handles to functions that require `AssetId`s

## Changelog

- Added ability to pass mutable references to `Handle`s as `AssetId`s

### CODE REVIEW

File: crates/bevy_asset/src/id.rs [hunk 1]
```diff
@@ -178,6 +178,13 @@ impl<A: Asset> From<&Handle<A>> for AssetId<A> {
     }
 }
 
+impl<A: Asset> From<&mut Handle<A>> for AssetId<A> {
```
Comment (alice-i-cecile) [inline crates/bevy_asset/src/id.rs:181]: Can we do something more general? It doesn't seem like we need mutability here, so we could probably do `AsRef<Handle<A>>`.

Comment (doonv) [reply crates/bevy_asset/src/id.rs:181]: Could you explain how I could implement that?

Comment (alice-i-cecile) [reply crates/bevy_asset/src/id.rs:181]:
```suggestion
impl<T: AsRef<Handle<A>>, A: Asset> From<T> for AssetId<A> {
```
Does this work?

Comment (alice-i-cecile) [reply crates/bevy_asset/src/id.rs:181]:
```suggestion
impl<T: Borrow<Handle<A>>, A: Asset> From<T> for AssetId<A> {
```

It appears that `Borrow` is preferred: https://www.reddit.com/r/rust/comments/xqrz14/asref_vs_borrow/

Comment (doonv) [reply crates/bevy_asset/src/id.rs]: Commited!

**Review by alice-i-cecile — APPROVED** [review]

**Comment by doonv** [comment]
It appears https://github.com/bevyengine/bevy/pull/10628 has already solved this problem. So I'm closing the PR.

**Pull request closed without merging**